src/lmio/components/User/utils/fetchAndStoreUser.js

```diff
@@ -0,0 +1,68 @@
+import i18next from 'i18next';
+
+// Compares array of IDs with data in localstorage
+export const matchCredentialId = (app, id, setData, cleanupTime) => {
+	const usernamesInLS = _getUsernamesFromLS('Credentials', cleanupTime);
+	if (!usernamesInLS.credentials || usernamesInLS.credentials.length === 0 || usernamesInLS.expiration <= Date.now()) {
+		_removeUsernamesFromLS();
+		_retrieveUserName(app, id, setData, cleanupTime);
+		return;
+	}
+	const found = usernamesInLS.credentials.find((item) => item.id === id);
+	if (!found) {
+		_retrieveUserName(app, id, setData, cleanupTime);
+	} else {
+		setData(found);
+	}
+};
+
+// asks the server for usernames, saves them to local storage and sets usernames to render
+const _retrieveUserName = async (app, cred_id, setData, cleanupTime) => {
+	const SeacatAuthAPI = app.axiosCreate('seacat-auth');
+	try {
+		let response = await CredentialsAPI.put(`idents`, [cred_id]);
```

Comment:
there is `CredentialsAPI`, but you declare SeacatAuthAPI (altough it should be SeaCatAuthAPI) - are you sure it has been tested?